src/holon/models/scenario.py

```diff
@@ -118,9 +152,12 @@ def delete_individualy(queryset):
         with transaction.atomic():
             # Delete polymorphic models individually
             # django-polymorphic can't handle deletion of mixed object types
+            from holon.models import Contract
+
             delete_individualy(self.assets)
             delete_individualy(self.gridconnection_set.all())
             delete_individualy(self.gridnode_set.all())
+            delete_individualy(Contract.objects.filter(actor__payload_id=self.id))
```

Comment:
Contracts could be collected the same way as assets to prevent an in-function import. But this works too